src/BaseFactory.php

```diff
@@ -78,9 +90,18 @@ public function with(string $relatedModelClass, string $relationshipName, int $t
     {
         $clone = clone $this;
 
-        $clone->relatedModelFactories = collect()->times($times, fn () => $this->getFactoryFromClassName($relatedModelClass));
+        $clone->relatedModelFactories = collect([
+            $relationshipName => collect()->times($times, fn() => $this->getFactoryFromClassName($relatedModelClass))
+        ]);
+
+        return $clone;
+    }
+
+    public function andWith(string $relatedModelClass, string $relationshipName, int $times = 1): self
```

Comment:
I'm wondering if the dev ux would be better without the "and" and only add more "with" methods to chain multiple relations. Guess it should work if we check for given relations in the with method?

Reply:
I also think it would be better DX, It would be a BC break though.
The following tests would fail: the_factory_is_immutable_when_adding_related_models would fail.
Would you realease a 2.0?

Reply:
I removed the `andWith` method and adapted `with`.